src/distribution/error.rs

```diff
@@ -90,13 +90,53 @@ pub struct ErrorInfo {
     /// MAY be empty.
     message: Option<String>,
 
-    #[serde(default, skip_serializing_if = "Option::is_none")]
+    #[serde(default, skip_serializing_if = "Option::is_none", with = "json_string")]
     #[builder(default = "None")]
     /// The detail field is OPTIONAL and MAY contain arbitrary JSON data providing information
     /// the client can use to resolve the issue.
     detail: Option<String>,
 }
 
+mod json_string {
+    use std::str::FromStr;
+
+    use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Option<String>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        use serde::de::Error;
+
+        let opt = Option::<serde_json::Value>::deserialize(deserializer)?;
+
+        if let Some(data) = opt {
+            let data = serde_json::to_string(&data).map_err(Error::custom)?;
+            return Ok(Some(data));
+        }
+
+        Ok(None)
+    }
+
+    pub fn serialize<S>(target: &Option<String>, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        use serde::ser::Error;
+
+        match target {
+            Some(data) => {
+                if let Ok(json_value) = serde_json::Value::from_str(data) {
+                    json_value.serialize(serializer)
+                } else {
+                    Err(Error::custom("invalid JSON"))
+                }
+            }
+            _ => unreachable!(),
```

Comment:
Can we erase `skip_serializing_if` and handle the skipping here?

Reply:
Yes we could, but I used the same approach as in `Config` [here](https://github.com/containers/oci-spec-rs/blob/5018f8e5b0355a82c08962cefa5ab07a05b930c6/src/image/config.rs#L253) and `serialize_as_map` function [here](https://github.com/containers/oci-spec-rs/blob/5018f8e5b0355a82c08962cefa5ab07a05b930c6/src/image/config.rs#L307). It looked to be more consistent this way. Should I fix this or not?

Reply:
It's not a very big deal, so let's leave it at that for this time. Thanks for the information.